xds/src/main/java/io/grpc/xds/internal/sds/ServerWrapperForXds.java

```diff
@@ -77,6 +77,7 @@ public Server start() throws IOException {
       throw new RuntimeException(ex);
     }
     delegate.start();
+    xdsServingStatusListener.onServing();
```

Comment:
Why not wait until `serverWatcher.onSuccess()` is called?

Reply:
It does. Line 96 sets the `settableFuture` in `onSuccess` and line 71 has `future.get()` which returns only when the future is set.